src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -351,11 +360,22 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
         interpolation_mesh.value_or(mesh).quadrature());
   }
 
-  using observation_registration_tags = tmpl::list<>;
+  using observation_registration_tags = tmpl::list<::Tags::DataBox>;
+
+  template <typename DbTagsList>
   std::pair<observers::TypeOfObservation, observers::ObservationKey>
-  get_observation_type_and_key_for_registration() const noexcept {
+  get_observation_type_and_key_for_registration(
+      const db::DataBox<DbTagsList>& box) const noexcept {
+    const std::optional<std::string> section_observation_key =
+        observers::get_section_observation_key<ArraySectionIdTag>(box);
+    ASSERT(section_observation_key != "Unused",
+           "The identifier 'Unused' is reserved to indicate that no "
+           "observations with this key will be contributed. Use a different "
+           "key, or change the identifier 'Unused' to something else.");
```

Comment:
Can we change these functions to return a `std::optional` instead of returning a dummy value?  The consumer `get_registration_observation_type_and_key` already uses an optional to store the result.

Reply:
Good idea! pushed a fixup